### ISSUE TITLE
kill task does not work if app id has  hierarchical path

### ISSUE THREAD

**Issue by lokeshverizon** [gambol99/go-marathon#111]
```
func (r *marathonClient) KillTask(taskID string, scale bool) (*Task, error) {
  ...
  appNamewithUnderScore := taskID[0:strings.LastIndex(taskID, ".")]
       
  // following line needs to be added. 
  appName := strings.Replace(appNamewithUnderScore, "_", "/", -1)
}
```

**Comment by timoreimann** [comment]
If I understand you correctly, the issue is about Marathon app names containing slashes that would end up in task IDs again. Looking at the [official REST documentation](https://mesosphere.github.io/marathon/docs/rest-api.html#post-v2-apps) (see the _identifier_ section), however, such Marathon IDs should be rejected in the first place as the slash is not among the set of allowed characters.

Am I missing something?

**Comment by lokeshverizon** [comment]
we can have app id with hierachical name. /group/project/application. 
Hence appid = /group/project/application
marathon is able to launch this application. this is converted to mesostask as:
group_project_application.<taskid>
with current API: it gives: group_project_application
which fails.

**Comment by lokeshverizon** [comment]
also in the doc you mentioned:     "id": "/product/service/myApp",

which is similar case as mine.

**Comment by lokeshverizon** [comment]
let me know if you need more info. i have local fix for this now. 
 appName := strings.Replace(appNamewithUnderScore, "_", "/", -1)
but would prefer to get it fix here.

**Comment by timoreimann** [comment]
Taking over this one.

@lokeshverizon, are you still interested in fixing this upstream? I'd be happy to review any pull requests you submit

**Comment by dmajere** [comment]
When you're trying to create marathon application, it will split your AppID into segments by "/", and then check if every segment is matching 
`^(([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9])\\.)*([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9])$|^(\\.|\\.\\.)$`

I've been thinking about implementing the same AppID validation in go-marathon library, so it can be run in app.Name() method and Application constructors. 
We have utils.validateID(), but it is not really validating ID.
I suggest changing
 `validateID(string) string`
 to 
`validateID(string) (bool, string)` and add actual validation in it.

**Comment by dmajere** [comment]
This one is solved with #150
